controllers/templatesync/template_sync.go

```diff
@@ -290,12 +317,24 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 
 			policyUserErrorsCounter.WithLabelValues(instance.Name, tName, "crd-error").Inc()
 
+			if isGkConstraintTemplate {
```

Comment:
What is there is more than one constraint or ConstraintTemplate in the policy-templates array?

Reply:
This assumes that if the ConstraintTemplate CRD isn't there, then there's no need for cleanup because Kubernetes ought to have taken care of that when the CRD disappeared? This is only being applied to ConstraintTemplates, not constraints.

Reply:
@dhaiducek yes, you are right. Could you please add that as a comment?

Reply:
> This should also be true for constraints as well.

(This is probably the 10th draft of this as I think about all of the flows 😂 ) There could be a weird scenario where a bunch of different constraints kinds are combined into a single policy. In that case, some CRDs might exist and others may not. We wouldn't want finalizer flapping in that case. For that, I might have to have a boolean on whether the finalizer is applied after going through the policy-templates...